dbcon/joblist/resourcemanager.cpp

```diff
@@ -411,22 +411,35 @@ bool ResourceManager::userPriorityEnabled() const
     return "Y" == val;
 }
 
+// Counts memory. This funtion doesn't actually malloc, just counts against two limits
+// totalUmMemLimit for overall UM counting and (optional) sessionLimit for a single session.
+// If both have space, return true.
 bool ResourceManager::getMemory(int64_t amount, boost::shared_ptr<int64_t> sessionLimit, bool patience)
```

Comment:
Moreover since we send down boost::shared_ptr<int64_t> and not its ref the code runs shared_ptr ctor that is an extra overhead. I don't see we really need this for the semantics but this particular note goes outside the scope of the patch.

Reply:
I agree on both counts. It probably could be improved with a ref and it is outside the scope.